src/seedu/addressbook/data/AddressBook.java

```diff
@@ -68,6 +68,16 @@ public UniquePersonList getAllPersons() {
         return new UniquePersonList(allPersons);
     }
 
+    /**
+     * Sorts the current UniquePersonList by mutating it based on Comparator
```

Comment:
Instead of mutating the `UniquePersonList` directly, try to mutate a copy of it to prevent changes to the original. This gives more freedom to keep track of some properties of the original `UniquePersonList` and implement more sort commands in the future - e.g. sort by recently added